documentation/code-generation.md

```diff
@@ -94,3 +94,70 @@ generator –r –v –l=temp –version=v11.2.0-beta C:/azure-rest-api-specs/sp
 ## Generate Azure SDK for Go profiles
 
 Take a look into the [profile generator documentation](https://github.com/Azure/azure-sdk-for-go/tree/main/eng/tools/profileBuilder)
+
+## Generate Go Mgmt SDK from Typespec
+### Prerequisites
+- [Node.js 18.x LTS](https://nodejs.org/en/download) or later
+- [Go 1.23.x](https://go.dev/doc/install) or later
+- [Git](https://git-scm.com/downloads)
+- Identify the `tspconfig.yaml` file for your package in the [Rest API Spec Repo](https://github.com/Azure/azure-rest-api-specs) and ensure there is a configuration for the Go SDK similar to that shown below and in [this example](https://github.com/Azure/azure-rest-api-specs/blob/b09c9ec927456021dc549e111fa2cac3b4b00659/specification/contosowidgetmanager/Contoso.Management/tspconfig.yaml#L40)
+
+     ```yaml
+     options:
+       "@azure-tools/typespec-go":
+         service-dir: "SERVICE_DIRECTORY_NAME"
+         package-dir: "PACKAGE_DIRECTORY_NAME"
```

Comment:
@tadelesh does the value of `package-dir` must start with `arm`? Any constraints like this should be documented in this page.

Reply:
yes. for mgmt. package, the package-dir should start with `arm`. this should be set according to the namespace review result.